src/sentry/uptime/grouptype.py

```diff
@@ -1,15 +1,289 @@
+        # TODO(epurkhiser): We should migrate the fingerprints over to match
+        # what the default fingerprint is.
+        return build_fingerprint(self.detector)
+
+    @override
+    def extract_dedupe_value(self, data_packet: DataPacket[UptimePacketValue]) -> int:
+        return int(data_packet.packet.check_result["scheduled_check_time_ms"])
+
+    @override
+    def evaluate(
+        self, data_packet: DataPacket[UptimePacketValue]
+    ) -> dict[DetectorGroupKey, DetectorEvaluationResult]:
+        result = super().evaluate(data_packet)
+
+        if not result:
+            return result
+
+        # Uptime does not use stateful detector value grouping
+        evaluation = result[None]
+
+        uptime_subscription = data_packet.packet.subscription
+        metric_tags = data_packet.packet.metric_tags
+
+        detector_issue_creation_enabled = features.has(
+            "organizations:uptime-detector-create-issues",
+            self.detector.project.organization,
+        )
+        issue_creation_flag_enabled = features.has(
+            "organizations:uptime-create-issues",
+            self.detector.project.organization,
+        )
+        restricted_host_provider_ids = options.get(
+            "uptime.restrict-issue-creation-by-hosting-provider-id"
+        )
+        host_provider_id = uptime_subscription.host_provider_id
+        host_provider_enabled = host_provider_id not in restricted_host_provider_ids
+
+        issue_creation_allowed = (
+            detector_issue_creation_enabled
+            and issue_creation_flag_enabled
+            and host_provider_enabled
+        )
+
+        # XXX(epurkhiser): We currently are duplicating the detector state onto
+        # the uptime_subscription when the detector changes state. Once we stop
+        # using this field we can drop this update logic.
+        #
+        # We ONLY do this when detector issue creation is enabled, otherwise we
+        # let the legacy uptime consumer handle this.
+        if detector_issue_creation_enabled:
+            if evaluation.priority == DetectorPriorityLevel.OK:
+                uptime_status = UptimeStatus.OK
+            elif evaluation.priority != DetectorPriorityLevel.OK:
+                uptime_status = UptimeStatus.FAILED
+
+            uptime_subscription.update(
+                uptime_status=uptime_status,
+                uptime_status_update_date=django_timezone.now(),
+            )
+
+        if not host_provider_enabled:
+            metrics.incr(
+                "uptime.result_processor.restricted_by_provider",
+                sample_rate=1.0,
+                tags={
+                    "host_provider_id": host_provider_id,
+                    **metric_tags,
+                },
+            )
+
+        result_creates_issue = isinstance(evaluation.result, IssueOccurrence)
+        result_resolves_issue = isinstance(evaluation.result, StatusChangeMessage)
+
+        if result_creates_issue:
```

Comment:
All these metrics seem like they should be provided by the platform

Reply:
Yeah I agree. But I don’t want to lose the metrics we have right now

Reply:
I can clean this all up later I just wanted to make sure we were not losing any metrics as we cut over from our legacy issue creation system

Reply:
It’s also valuable for us to have various metric tags. Probably something worth talking about for the handler APIs